### ISSUE TITLE
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

### ISSUE THREAD

**Issue by jz451** [facebook/prophet#1694]
This is my first time using fbprophet in Python and right out of the gate it's not working giving me the following error. The error even occurred when creating a test time series dataset in Excel as I figured it had to do with there being multiple of the same date. I also had some issues when installing fbprophet and its dependencies so I don't know if that's contributing. Not sure what to do because the df.any() and df.all() tests return True values for ds and y columns.

```
from fbprophet import Prophet
m = Prophet()
m.fit(df)
```

This returns the error:

```
ValueError                                Traceback (most recent call last)
<ipython-input-200-b474a6bc2b99> in <module>
      1 from fbprophet import Prophet
      2 m = Prophet()
----> 3 m.fit(df)

~\Anaconda3\lib\site-packages\fbprophet\forecaster.py in fit(self, df, **kwargs)
   1109         self.history_dates = pd.to_datetime(pd.Series(df['ds'].unique(), name='ds')).sort_values()
   1110 
-> 1111         history = self.setup_dataframe(history, initialize_scales=True)
   1112         self.history = history
   1113         self.set_auto_seasonalities()

~\Anaconda3\lib\site-packages\fbprophet\forecaster.py in setup_dataframe(self, df, initialize_scales)
    320         df['t'] = (df['ds'] - self.start) / self.t_scale
    321         if 'y' in df:
--> 322             df['y_scaled'] = (df['y'] - df['floor']) / self.y_scale
    323 
    324         for name, props in self.extra_regressors.items():

~\Anaconda3\lib\site-packages\pandas\core\ops\__init__.py in wrapper(left, right)

~\Anaconda3\lib\site-packages\pandas\core\ops\__init__.py in na_op(x, y)

~\Anaconda3\lib\site-packages\pandas\core\computation\expressions.py in evaluate(op, a, b, use_numexpr)
    226     op_str = _op_str_mapping[op]
    227     if op_str is not None:
--> 228         use_numexpr = use_numexpr and _bool_arith_check(op_str, a, b)
    229         if use_numexpr:
    230             return _evaluate(op, op_str, a, b)  # type: ignore

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

**Comment by p0ete** [comment]
I have the same issue.

**Comment by allendred** [comment]
I have the same issue.

**Comment by bletham** [comment]
Is there any chance you can share the dataframe you are passing in to `fit` so I can try to replicate?

The actual error seems to be coming from code involving numexpr. This is an optional backend for pandas. I don't know a whole lot about it myself, but it looks like `use_numexpr` is evaluating to True in the line raising the error (if you're able to drop into a debugger and check that, that'd be great).

Could you try disabling numexpr and seeing if that helps? It looks like this will do it:
```Python
import pandas as pd
pd.set_option('compute.use_numexpr', False)
```

**Comment by noor-zehra** [comment]
I am getting the exact same error. Unfortunately disabling numexpr does not resolve the issue.
Here is the dataframe I am using to fit:

[fbprophet.parquet.zip](https://github.com/facebook/prophet/files/5345998/fbprophet.parquet.zip)

**Comment by bletham** [comment]
@noor-zehra Thanks for sharing the dataframe. When I run this code I do not run into the error:
```Python
import pandas as pd
from fbprophet import Prophet

df = pd.read_parquet('fbprophet.parquet')

m = Prophet()
m.fit(df)
```
This is with pandas==1.1.2, numexpr==2.7.1, and fbprophet==0.7.1.

What versions of pandas, numexpr, and fbprophet is everyone using that is running into this issue?

The error is coming from pandas, in a line where it is trying to determine if it should use numexpr or not. It looks like on my pandas install it is using numexpr by default, so I'm not sure why that line seems to be working for me but producing an error for others, other than perhaps a change in versions.
```
>>> pd.describe_option('compute.use_numexpr')
compute.use_numexpr : bool
    Use the numexpr library to accelerate computation if it is installed,
    the default is True
    Valid values: False,True
    [default: True] [currently: True]
```

**Comment by noor-zehra** [comment]
So I was using pandas==1.1.3, numexpr==2.7.1, and fbprophet==0.7.1.

Because when I installed fbprophet, pandas 1.1.3 was automatically installed. 
Downgrading pandas to 1.1.2 solved my problem. Perhaps this is some issue coming from some latest change in pandas. 

Thanks a lot.

**Comment by bletham** [comment]
Thanks for the update. I'm still a bit unsure what might be happening here. I think the numexpr is probably a red herring, from your result of downgrading pandas fixing the issue it seems like it could be something in how pandas is configured. But I'm really not sure what because I upgraded my pandas to both 1.1.3 and 1.1.4 and did not reproduce the issue using the code that I posted above. In any case, glad that you got it working.
